### ISSUE TITLE
SpanData::instrumentation_library attributes not being exported?

### ISSUE THREAD

**Issue by shaun-cox** [open-telemetry/opentelemetry-rust#1176]
The [`InstrumentationLibrary`](https://github.com/open-telemetry/opentelemetry-rust/blob/dfeac078ff7853e7dc814778524b93470dfa5c9c/opentelemetry-sdk/src/export/trace.rs#L94) carries attributes.  However, I don't see these being exported.  Am I correct in assuming that code needs to be added roughly [here](https://github.com/open-telemetry/opentelemetry-rust/blob/dfeac078ff7853e7dc814778524b93470dfa5c9c/opentelemetry-proto/src/transform/traces.rs#L52C47-L52C47) to make that happen, or are they normalized and exported only once for each instrumentation library somewhere else?  (Or am I incorrect in assuming that they should be exported at all?)

**Comment by cijothomas** [comment]
>  (Or am I incorrect in assuming that they should be exported at all?

I think so! The instrumentation library (along with Resource) should be available for Exporters, but how they are exported depends on the Exporter itself. OTLP has well-known place to put Instrumentation Library, Resource, but that is not the case with many non-OTLP exporters. A lot of exporters would be simply ignoring things it does not have interest in.

Apart from OTLP, the spec has clarified how to export things to sytems like Zipkin : https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/zipkin.md

In short, I don't think we need to do anything here as long as exporters have access to Resource/Instrumentation Library (which seem to be the case.)

**Comment by shaun-cox** [comment]
Maybe I'm still confused, but it seems there are attributes in [`InstrumentationScope`](https://github.com/open-telemetry/opentelemetry-proto/blob/c4dfbc51f3cd4089778555a2ac5d9bc093ed2956/opentelemetry/proto/common/v1/common.proto#L79) protobuf definitions.

But I don't see where they are being populated in the conversion to protobuf for [`From<InstrumentationLibrary>`](https://github.com/open-telemetry/opentelemetry-rust/blob/dfeac078ff7853e7dc814778524b93470dfa5c9c/opentelemetry-proto/src/transform/common.rs#L147-L150).  (I didn't include this link before, but hopefully its clearer now.)

(I'm using the OTLP exporter, but attributes of my `Tracer`s are not showing up in jaeger.)

**Comment by lalitb** [comment]
Yes agree. Attributes from `InstrumentationLibrary` should be included in the conversion to `InstrumentationScope`

**Comment by cijothomas** [comment]
> Maybe I'm still confused, but it seems there are attributes in [`InstrumentationScope`](https://github.com/open-telemetry/opentelemetry-proto/blob/c4dfbc51f3cd4089778555a2ac5d9bc093ed2956/opentelemetry/proto/common/v1/common.proto#L79) protobuf definitions.
> 
> But I don't see where they are being populated in the conversion to protobuf for [`From<InstrumentationLibrary>`](https://github.com/open-telemetry/opentelemetry-rust/blob/dfeac078ff7853e7dc814778524b93470dfa5c9c/opentelemetry-proto/src/transform/common.rs#L147-L150). (I didn't include this link before, but hopefully its clearer now.)
> 
> (I'm using the OTLP exporter, but attributes of my `Tracer`s are not showing up in jaeger.)

Got it.! Sorry I didn't notice this was referring to OTLP Exporter! Yes it looks like a gap that OTLP Exporter is not populating them.
